Examples/Io/EDM4hep/src/EDM4hepUtil.cpp

```diff
@@ -0,0 +1,305 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Io/EDM4hep/EDM4hepUtil.hpp"
+
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/MultiTrajectory.hpp"
+#include "Acts/EventData/MultiTrajectoryHelpers.hpp"
+#include "Acts/Utilities/Helpers.hpp"
+#include "ActsExamples/Digitization/MeasurementCreation.hpp"
+#include "ActsExamples/EventData/Index.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/Validation/TrackClassification.hpp"
+
+#include "edm4hep/TrackState.h"
+
+namespace ActsExamples {
+
+ActsFatras::Particle EDM4hepUtil::readParticle(
+    edm4hep::MCParticle from, MapParticleIdFrom particleMapper) {
+  ActsFatras::Barcode particleId = particleMapper(from);
+
+  ActsFatras::Particle to(particleId, (Acts::PdgParticle)from.getPDG(),
+                          from.getCharge() * Acts::UnitConstants::e,
+                          from.getMass() * Acts::UnitConstants::GeV);
+
+  // TODO do we have that in EDM4hep?
+  // particle.setProcess(static_cast<ActsFatras::ProcessType>(data.process));
+
+  to.setPosition4(from.getVertex()[0] * Acts::UnitConstants::mm,
```

Comment:
Are the input units fixed from edm4hep? Otherwise we'd have to divide by their unit constants first.

Reply:
according to their doc the vertex should be in `mm` https://github.com/key4hep/EDM4hep/blob/master/edm4hep.yaml#L133